utils/exceptions.hpp

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include <QException>
+#include <QMessageBox>
+
+namespace jbrush
+{
+// Inherits from QException and allows the client to specify a unique string message that
+// is displayed in Qt's debugging console when raised
+class GeneralException : public QException
+{
+private:
+    QString message;
+public:
+    GeneralException(QString m) { message = m; }
+    void raise() const override { throw *this; }
+    GeneralException* clone() const override { return new GeneralException(*this); }
+
+    // Return the error message assigned to this exception
+    QString errorMessage () const { return message; }
+
+    // Causes a message box to appear describing the error that occurred
+    // Good for informing the user of errors they need to be aware of
+    void errorWindow () const
+    {
+       QMessageBox::critical(nullptr, "Error", message);
+    }
+};
+
+// Throw this exception if a file was not opened properly
+class BadFile : public GeneralException
```

Comment:
It should require a ``QFile``. Something similar to below should be what the error message is.
```
"shapes.txt" is a bad file.
```